### PR TITLE
Budget App: Rails Capstone Project

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Creating a Budget App with Rails
In this capstone project, I built a mobile web app that allows users to manage their budgets.

## 💻 Development Process
- Setup a new Rails app with Postgres database
- Created user authentication system with `devise`
- Created models and database schema based on [this ERD diagram](https://user-images.githubusercontent.com/72640048/235965558-f760c17d-6e5f-44aa-b929-152806157b87.png)
- Added splash screen and authentication forms
- Created the following pages:
  - **Homepage**- displays the list of all categories
  - **Transactions**- displays the list of all transactions for a category
- Created the following forms:
  - `category#new`- to add a new category
  - `expense#new`- to add a new expense
- Implemented `#create` and `#destroy` actions for categories and expenses 
- Added side menu
- Added unit tests for models' validations and associations
- Added request specs to test HTTP requests and responses
- Added feature specs to test user interactions
- Deployed the app on **Render**

## 🚀 Live Demo
You can visit the live website here:
- [CoinTrackr - Budget App](https://cointrackr.onrender.com/)

## 🎥 Presentation
This is my project presentation:
- [Presentation Video](https://www.loom.com/share/4eb28da6078245d4ad3babcae7446406)

*Thanks for reviewing my project. I'd love to hear your feedback and suggestions.*

### CODE REVIEW

File: app/views/shared/_header.html.erb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    <% end %>
+  </div>
+</div>
+<% end %>
+
+<script>
+function toggleSideMenu() {
+  const menu = document.querySelector(".side-menu");
+  menu.classList.toggle('show');
+}
```
Comment (VICTOR-CHUKWU) [inline app/views/shared/_header.html.erb:68]: - [ ] nice implementation for the sidebar menu :clap:

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb:5]: - [ ] it would have been so nice if you added confirmable so that a user can confirm his email :+1:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @IndieCoderMM ,
# STATUS: APPROVED! 🥇 💯 
- [ ] This is actually one of the best project I have reviewed. Bravo!!! 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/models/user.rb (hunk 2 above)
Comment (IndieCoderMM) [reply app/models/user.rb:5]: Thanks for your kind review. 😃
I'll try to implement this feature in the next update.

**Pull request merged**